src/stores/game.js

```diff
@@ -236,7 +238,12 @@ export const useGameStore = defineStore('game', {
       this.currentMatch = newGame.currentMatch ?? this.currentMatch;
       this.p0Rematch = newGame.p0Rematch ?? null;
       this.p1Rematch = newGame.p1Rematch ?? null;
+      this.rematchGameId = newGame.rematchGame ?? null;
       this.gameIsOver = newGame.gameIsOver ?? false;
+      this.status = newGame.status ?? GameStatus.ARCHIVED;
+    
+      const gameHistoryStore = useGameHistoryStore();
+      gameHistoryStore.gameStates = newGame.gameStates ?? [];
```

Comment:
setting `rematchGameId`, `status`, and `gameStates` on every call to `updateGame()` to keep them up to date with each update when navigating across states during replays, instead of relying on only specific socket events to update those properties